Update needed for Sphinx 8.0

### Describe the bug

The CI build of the SciPy docs is [failing](https://github.com/scipy/scipy/issues/21323) with the recently updated Sphinx 8.0.2.  Initially the problem was coming from `jupyterlite_sphinx` (see https://github.com/jupyterlite/jupyterlite-sphinx/issues/197).  A current [pull request](https://github.com/jupyterlite/jupyterlite-sphinx/pull/199) there fixes the error that occurs in `jupyterlite_sphinx`, but the build of the SciPy docs fails again in `myst_nb` code:

```
[...snip...]
reading sources... [ 96%] tutorial/interpolate/interp_transition_guide
Traceback (most recent call last):
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/cmd/build.py", line 337, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/application.py", line 378, in build
    self.builder.build_update()
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 296, in build_update
    self.build(to_build,
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 317, in build
    updated_docnames = set(self.read())
                           ^^^^^^^^^^^
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 424, in read
    self._read_serial(docnames)
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 476, in _read_serial
    self.read_doc(docname)
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 535, in read_doc
    publisher.publish()
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/docutils/core.py", line 234, in publish
    self.document = self.reader.read(self.source, self.parser,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/io.py", line 106, in read
    self.parse()
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/myst_nb/sphinx_.py", line 85, in parse
    nb_reader = create_nb_reader(document_path, md_config, nb_config, inputstring)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/myst_nb/core/read.py", line 71, in create_nb_reader
    if path.endswith(suffix):
       ^^^^^^^^^^^^^
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/util/_pathlib.py", line 96, in __getattr__
    warnings.warn(_MSG, RemovedInSphinx90Warning, stacklevel=2)
sphinx.deprecation.RemovedInSphinx90Warning: Sphinx 9 will drop support for representing paths as strings. Use "pathlib.Path" or "os.fspath" instead.

Exception occurred:
  File "/home/warren/py3.12.4/lib/python3.12/site-packages/sphinx/util/_pathlib.py", line 96, in __getattr__
    warnings.warn(_MSG, RemovedInSphinx90Warning, stacklevel=2)
sphinx.deprecation.RemovedInSphinx90Warning: Sphinx 9 will drop support for representing paths as strings. Use "pathlib.Path" or "os.fspath" instead.
The full traceback has been saved in /tmp/sphinx-err-1o84nyhg.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:114: html-build] Error 2
```

The error occurs in 'core/read.py' at line 71:

https://github.com/executablebooks/MyST-NB/blob/6ce30cd41fa82543e0f315ac8bbee82669b0cc82/myst_nb/core/read.py#L71

If I follow the example of the [`jupyterlite_sphinx` pull request](https://github.com/jupyterlite/jupyterlite-sphinx/pull/199), and change that line to:

```
         if path.suffix == suffix:
```

the build of the SciPy docs completes successfully.

### Reproduce the bug

See above.

### List your environment

_No response_

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -21,11 +21,11 @@ jobs:
         include:
           # Just check the other platforms once
           - os: windows-latest
-            python-version: "3.10"
+            python-version: "3.12"
             sphinx: "~=8.0"
             myst-parser: "~=4.0"
           - os: macos-latest
-            python-version: "3.10"
+            python-version: "3.12"
             sphinx: "~=8.0"
             myst-parser: "~=4.0"
           # Oldest known-compatible dependencies
@@ -33,6 +33,11 @@ jobs:
             python-version: "3.9"
             sphinx: "==5.0.0"
             myst-parser: "==1.0.0"
+          # Mid-range dependencies
+          - os: ubuntu-latest
+            python-version: "3.11"
+            sphinx: "==7.0.0"
+            myst-parser: "==2.0.0"
           # Newest known-compatible dependencies
           - os: ubuntu-latest
             python-version: "3.12"

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -23,11 +23,11 @@ jobs:
           - os: windows-latest
             python-version: "3.10"
             sphinx: "~=8.0"
-            myst-parser: "~=2.0"
+            myst-parser: "~=4.0"
           - os: macos-latest
             python-version: "3.10"
             sphinx: "~=8.0"
-            myst-parser: "~=2.0"
+            myst-parser: "~=4.0"
           # Oldest known-compatible dependencies
           - os: ubuntu-latest
             python-version: "3.9"
@@ -36,8 +36,8 @@ jobs:
           # Newest known-compatible dependencies
           - os: ubuntu-latest
             python-version: "3.12"
-            sphinx: "==8.0.0"
-            myst-parser: "==2.0.0"
+            sphinx: "==8.0.2"
+            myst-parser: "==4.0.0"
 
     runs-on: ${{ matrix.os }}
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -16,17 +16,17 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         python-version: ["3.9", "3.10", "3.11", "3.12"]
-        sphinx: [">=5,<8"]  # Newest Sphinx (any)
+        sphinx: [">=5,<9"]  # Newest Sphinx (any)
         myst-parser: [">=1,<3"]  # Newest MyST Parser (any)
         include:
           # Just check the other platforms once
           - os: windows-latest
             python-version: "3.10"
-            sphinx: "~=7.0"
+            sphinx: "~=8.0"
             myst-parser: "~=2.0"
           - os: macos-latest
             python-version: "3.10"
-            sphinx: "~=7.0"
+            sphinx: "~=8.0"
             myst-parser: "~=2.0"
           # Oldest known-compatible dependencies
           - os: ubuntu-latest
@@ -36,7 +36,7 @@ jobs:
           # Newest known-compatible dependencies
           - os: ubuntu-latest
             python-version: "3.12"
-            sphinx: "==7.0.0"
+            sphinx: "==8.0.0"
             myst-parser: "==2.0.0"
 
     runs-on: ${{ matrix.os }}

--- a/myst_nb/core/read.py
+++ b/myst_nb/core/read.py
@@ -68,7 +68,7 @@ def create_nb_reader(
     # we check suffixes ordered by longest first, to ensure we get the "closest" match
     iterator = sorted(readers.items(), key=lambda x: len(x[0]), reverse=True)
     for suffix, (reader, reader_kwargs, commonmark_only) in iterator:
-        if path.endswith(suffix):
+        if path.suffix == suffix:
             if isinstance(reader, str):
                 # attempt to load the reader as an object path
                 reader = import_object(reader)

--- a/myst_nb/core/read.py
+++ b/myst_nb/core/read.py
@@ -68,7 +68,7 @@ def create_nb_reader(
     # we check suffixes ordered by longest first, to ensure we get the "closest" match
     iterator = sorted(readers.items(), key=lambda x: len(x[0]), reverse=True)
     for suffix, (reader, reader_kwargs, commonmark_only) in iterator:
-        if path.suffix == suffix:
+        if Path(path).suffix == suffix:
             if isinstance(reader, str):
                 # attempt to load the reader as an object path
                 reader = import_object(reader)

--- a/tox.ini
+++ b/tox.ini
@@ -16,12 +16,13 @@ envlist = py311-sphinx7
 [testenv]
 usedevelop = true
 
-[testenv:py{39,310,311,312}-sphinx{5,6,7}]
+[testenv:py{39,310,311,312}-sphinx{5,6,7,8}]
 extras = testing
 deps =
     sphinx5: sphinx>=5,<6
     sphinx6: sphinx>=6,<7
     sphinx7: sphinx>=7,<8
+    sphinx8: sphinx>=8,<9
 commands = pytest {posargs}
 
 [testenv:docs-{update,clean}]

